### PR TITLE
Added extension for ObservableType bindings

### PR DESCRIPTION
**Pull request by ThasianX**

### CODE REVIEW

File: DrX/Sources/Core/ObservableType+Relay.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+extension ObservableType {
+    func bind<R: RelayType>(to relay: R) -> Disposable where R.Element == Element {
+        bind(to: relay.asObserver())
+    }
+    
+    func bind<R: RelayType>(to relay: R) -> Disposable where R.Element == Element? {
+        map(Optional.some)
+            .bind(to: relay.asObserver())
+    }
```
Comment (rpwachowski) [inline DrX/Sources/Core/ObservableType+Relay.swift:13]: These generic functions make the extension functions to take specifically `AnyRelay` unnecessary—try out removing those definitions and making sure they still work.

Comment (ThasianX) [reply DrX/Sources/Core/ObservableType+Relay.swift:13]: holy shit the thing that threw me off was u using "to take" instead of "that take"

**Comment by powerje** [comment]
Please bump the podspec version, it should be a minor bump.

**Comment by powerje** [comment]
Build is failing for CI, Carthage can't build Nimble:
```
*** Building scheme "Quick-iOS" in Quick.xcworkspace
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -workspace /Users/travis/build/DuetHealth/DrX/Carthage/Checkouts/Quick/Quick.xcworkspace -scheme Quick-iOS -configuration Release -derivedDataPath /Users/travis/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3.1_11C505/Quick/v2.2.0 -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/17/5mc7816d3mndxjqgplq6057w0000gn/T/Quick SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in /Users/travis/build/DuetHealth/DrX/Carthage/Checkouts/Quick)
This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/17/5mc7816d3mndxjqgplq6057w0000gn/T/carthage-xcodebuild.EYw2jC.log
The command "carthage update --platform iOS" failed and exited with 1 during .
```

But works locally for me 🤷‍♀

**Comment by powerje** [comment]
I'm going to say since `swift build` works we can just merge this, it's failing due to the maximum number of log in attempts being exceeded by Travis

**Pull request merged**